src/gerbil/runtime/mop.ss

```diff
@@ -17,65 +17,121 @@ namespace: #f
+;;  2  ##type-name                 : Symbol
+;;  3  ##type-flags                : Fixnum
+;;  4  ##type-super                : (OrFalse StructTypeDescriptor)
+;;  5  ##type-fields               : (Vector [Symbol Fixnum default-value] ...)
+;;  6  class-type-precedence-list  : (List TypeDescriptor) ; doesn't contain the class itself
+;;  7  class-type-slot-vector      : (Vector Symbol) ; first is always ##type
+;;  8  class-type-slot-table       : (Table (Or Symbol Keyword) -> Fixnum)
+;;  9  class-type-properties       : AList ; NB: not PList, despite the name "properties"
+;; 10  class-type-constructor      : (OrFalse Symbol)
+;; 11  class-type-methods          : (Table Symbol -> Function)
 ;;
 ;; The ##type-fields contains 3 entries by Gambit structure field, field name, flags and default value
 ;;
 ;; Standard "properties" keys:
-;; direct-slots:   (List Symbol)           list of direct slots in order, for inheritance
-;; direct-supers:  (List TypeDescriptor)   list of direct supers in order, for introspection only
-;; struct:         Bool                    is this a struct? i.e. only single inheritance for structs
-;; final:          Bool                    is this class final? i.e. no subclasses allowed
-;; equal:          (Or Bool (List Symbol)  all new slots will be compared during objects equal?
-;; print:          (Or Bool (List Symbol)  all new slots will be printed when printing an object
-;; transparent:    Bool                    is this class transparent? i.e. equal and print all slots
-
-(def (type-id klass)
+;; direct-slots:   (List Symbol)            list of direct slots in order, for inheritance
+;; direct-supers:  (List TypeDescriptor)    list of direct supers in order, for introspection only
+;; struct:         Bool                     is this a struct? i.e. only single inheritance for structs
+;; final:          Bool                     is this class final? i.e. no subclasses allowed
+;; equal:          (Or Bool (List Symbol))  all new slots will be compared during objects equal?
+;; print:          (Or Bool (List Symbol))  all new slots will be printed when printing an object
+;; transparent:    Bool                     is this class transparent? i.e. equal and print all slots
+
+;; These type flags are defined by Gambit itself
+(def type-flag-opaque 1) ;; if opaque, instances are only equal? if eq?
+(def type-flag-extensible 2) ;; if 0, no type extension by Gambit
+(def type-flag-macros 4) ;; if 1, Gambit generates macros for accessors -- irrelevant for Gerbil
+(def type-flag-concrete 8) ;; aka constructor -- has direct instances -- always set for Gerbil
+(def type-flag-id 16) ;; if set we can compare by ##type-id and not just by eq? -- always set for Gerbl
+;; These class type flags are Gerbil extensions
+(def class-type-flag-struct 256) ;; precedence-list always tail of subclass's precedence-list
+(def class-type-flag-sealed 512) ;; no new changes, subclasses or method definitions (implies final)
```

Comment:
let's leave some more bits below? I'd feel better if we left at least 10, so let's start at 1024.